R/data-column.R

```diff
@@ -0,0 +1,52 @@
+WITH_DIMENSION_EXTENSION <- "OSPSuite.Core.Domain.WithDimensionExtensions"
+WITH_DISPLAY_UNIT_EXTENSION <- "OSPSuite.Core.Domain.WithDisplayUnitExtensions"
+
+#' @title DataColumn
+#' @docType class
+#' @description  An object typically holding observed data
+#' @format NULL
+DataColumn <- R6::R6Class(
+  "DataColumn",
+  inherit = DotNetWrapper,
+  cloneable = FALSE,
+  active = list(
+    #' @field values Returns the values defined in the column
+    values = function(value) {
+      private$wrapReadOnlyProperty("ValuesAsArray", value)
+    },
+    name = function(value){
+      private$wrapReadOnlyProperty("Name", value)
+    },
+    #' @field unit The base unit in which the quantity value is defined (Read-Only)
+    unit = function(value) {
+      private$.unit <- private$wrapExtensionMethodCached(WITH_DIMENSION_EXTENSION, "BaseUnitName", "unit", private$.unit, value)
+      return(private$.unit)
+    },
+    #' @field displayUnit The unit in which the quantity value is usually displayed (Read-Only)
+    displayUnit = function(value) {
+      private$wrapExtensionMethod(WITH_DISPLAY_UNIT_EXTENSION, "DisplayUnitName", "displayUnit", value)
+    },
+    #' @field dimension The dimension in which the quantity is defined  (Read-Only)
+    dimension = function(value) {
+      private$.dimension <- private$wrapExtensionMethodCached(WITH_DIMENSION_EXTENSION, "DimensionName", "dimension", private$.dimension, value)
+      return(private$.dimension)
+    }
+  ),
+  public = list(
+    #' @description
+    #' Print the object to the console
+    #' @param ... Rest arguments.
+    print = function(...) {
+      if (self$unit == "") {
+        private$printLine(self$name)
+      } else {

```

Comment:
@msevestre 
I would not make this differentiation. `[]` is assumed to be the unit of `Dimensionless` and is used like this e.g. in excel files when reading the data. From my perspective, it would be ok to print something like

> Data  name, []

Reply:
I think we are removing it when printing a quantity value etc so I just wanted to ber consistent

Reply:
see 
```
 #' @description
    #' Print the the value and unit of the quantity
    #' @param  caption Text to prepend to the value
    printQuantityValue = function(caption) {
      if (self$unit == "") {
        private$printLine(caption, formatNumerics(self$value))
      } else {
        private$printLine(caption, paste0(formatNumerics(self$value), " [", self$unit, "]"))
      }
    },
```

Reply:
All right then.